ambry-store/src/test/java/com.github.ambry.store/DefaultCompactionPolicyTest.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Time;
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.Properties;
+import org.junit.Test;
+
+import static com.github.ambry.store.CompactionPolicyTest.*;
+
+
+public class DefaultCompactionPolicyTest {
+
+  private static long DEFAULT_MAX_BLOB_SIZE = CAPACITY_IN_BYTES / 100;
+  // the properties that will used to generate a StoreConfig. Clear before use if required.
+  private final Properties properties = new Properties();
+  private final Time time = new MockTime();
+  private StoreConfig config;
+  private MockBlobStore blobStore;
+  private long messageRetentionTimeInMs;
+  private MockBlobStoreStats mockBlobStoreStats;
+  private CompactionPolicy compactionPolicy = null;
+
+  /**
+   * Instantiates {@link CompactionPolicyTest} with the required cast
+   * @throws InterruptedException
+   */
+  public DefaultCompactionPolicyTest() throws InterruptedException {
+    Pair<MockBlobStore, StoreConfig> initState =
+        CompactionPolicyTest.initializeBlobStore(properties, time, -1, -1, DEFAULT_MAX_BLOB_SIZE);
+    this.config = initState.getSecond();
+    this.blobStore = initState.getFirst();
+    this.mockBlobStoreStats = blobStore.getBlobStoreStats();
+    this.messageRetentionTimeInMs =
+        config.storeDeletedMessageRetentionDays * Time.HoursPerDay * Time.SecsPerHour * Time.MsPerSec;
+    this.compactionPolicy = new DefaultCompactionPolicy(config, time);
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)}
+   * @throws StoreException
+   * @throws InterruptedException
+   */
+  @Test
+  public void testGetCompactionDetailsTest() throws StoreException, InterruptedException {
+    // null logSegmentsNotInJournal
```

Comment:
Could this comment explain why this is set to null here (and then set again a few lines below)?